### PR TITLE
feat(editor): create new workflows page

### PR DESCRIPTION
**Pull request by alexgrozav**

### CODE REVIEW

File: packages/editor-ui/src/components/WorkflowCard.vue [hunk 1]
```diff
@@ -1,66 +1,240 @@
...
+				}
+
+				// Reset tab title since workflow is deleted.
+				this.$showMessage({
+					title: this.$locale.baseText('mainSidebar.showMessage.handleSelect1.title'),
+					type: 'success',
+				});
+			}
+		},
+		onActiveChange (data: { id: string, active: boolean }) {
```
Comment (mutdmour) [inline packages/editor-ui/src/components/WorkflowCard.vue:199]: we should move this to workflow activator level.. that's where we are making the api request now

File: packages/editor-ui/src/components/WorkflowCard.vue [hunk 2]
```diff
@@ -1,66 +1,240 @@
...
+		},
+	},
+	methods: {
+		async onClick(event?: PointerEvent) {
+			if (event) {
+				if ((this.$refs.activator as Vue)?.$el.contains(event.target as HTMLElement)) {
+					return;
+				}
+
+				if (event.metaKey || event.ctrlKey) {
```
Comment (mutdmour) [inline packages/editor-ui/src/components/WorkflowCard.vue:149]: ❤️ thanks for keeping this

File: packages/editor-ui/src/components/forms/ResourceFiltersDropdown.vue [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		ResourceSharingFilterSelect,
+	},
+	props: {
+		value: {
+			type: Object,
+			default: () => ({}),
+		},
+		keys: {
+			type: Array,
+			default: (): string[] => [],
```
Comment (mutdmour) [inline packages/editor-ui/src/components/forms/ResourceFiltersDropdown.vue:78]: would be great to add types here... `as PropType<whatever>`

File: packages/editor-ui/src/components/forms/ResourceSharingFilterSelect.ee.vue [hunk 4]
```diff
@@ -0,0 +1,65 @@
+<template>
+	<n8n-select
```
Comment (mutdmour) [inline packages/editor-ui/src/components/forms/ResourceSharingFilterSelect.ee.vue:2]: this is not the same as n8n-user-select ?

File: packages/editor-ui/src/plugins/i18n/locales/en.json [hunk 5]
```diff
@@ -1114,6 +1113,40 @@
...
+	"workflows.filters.ownedBy": "Owned by",
+	"workflows.filters.sharedWith": "Shared with",
+	"workflows.filters.apply": "Apply filters",
+	"workflows.filters.reset": "Reset all",
+	"workflows.filters.active": "Some workflows may be hidden since filters are applied.",
+	"workflows.filters.active.reset": "Remove filters",
+	"workflows.sort.lastUpdated": "Sort by last updated",
+	"workflows.sort.lastCreated": "Sort by last created",
+	"workflows.sort.nameAsc": "Sort by name (A-Z)",
+	"workflows.sort.nameDesc": "Sort by name (Z-A)",
```
Comment (mutdmour) [inline packages/editor-ui/src/plugins/i18n/locales/en.json:1139]: a few of these don't seem to be used

File: packages/editor-ui/src/components/layouts/ResourcesListLayout.vue [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+			this.$telemetry.track('User changed credentials sub view', {
+				sub_view: this.getTelemetrySubview(),
+			});
+		},
+		sendSortingTelemetry() {
+			this.$telemetry.track('User changed sorting in cred list', {
+				sub_view: this.getTelemetrySubview(),
+				sorting: this.sortBy,
+			});
+		},
```
Comment (mutdmour) [inline packages/editor-ui/src/components/layouts/ResourcesListLayout.vue:313]: hmm seems like these should be refactored out

File: packages/editor-ui/src/components/layouts/ResourcesListLayout.vue [hunk 7]
```diff
@@ -0,0 +1,405 @@
...
+</template>
+
+<script lang="ts">
+import {showMessage} from '@/components/mixins/showMessage';
+import {IUser} from '@/Interface';
+import mixins from 'vue-typed-mixins';
+
+import PageViewLayout from "@/components/layouts/PageViewLayout.vue";
+import PageViewLayoutList from "@/components/layouts/PageViewLayoutList.vue";
+import CredentialCard from "@/components/CredentialCard.vue";
```
Comment (mutdmour) [inline packages/editor-ui/src/components/layouts/ResourcesListLayout.vue:126]: is this still used?

File: packages/editor-ui/src/components/layouts/ResourcesListLayout.vue [hunk 8]
```diff
@@ -0,0 +1,405 @@
...
+
+type IResourceKeyType = 'credentials' | 'workflows';
+
+const filterKeys = ['ownedBy', 'sharedWith'];
+
+export default mixins(
+	showMessage,
+	debounceHelper,
+).extend({
+	name: 'ResourcesListLayout',
```
Comment (mutdmour) [inline packages/editor-ui/src/components/layouts/ResourcesListLayout.vue:159]: reminder to write in kebab-case

File: packages/editor-ui/src/components/layouts/ResourcesListLayout.vue [hunk 9]
```diff
@@ -0,0 +1,405 @@
...
+			const filterValues: Array<string[] | string | boolean | null> = [];
+
+			Object.keys(filters).forEach((key) => {
+				if (filters[key]) {
+					filtersSet.push(key);
+					filterValues.push(key === 'search' ? null : filters[key]);
+				}
+			});
+
+			this.$telemetry.track('User set filters in cred list', {
```
Comment (mutdmour) [inline packages/editor-ui/src/components/layouts/ResourcesListLayout.vue:336]: should be refactored out..

File: packages/editor-ui/src/store.ts [hunk 10]
```diff
@@ -185,12 +204,20 @@ export const store = new Vuex.Store({
 			if (index === -1) {
 				state.activeWorkflows.push(workflowId);
 			}
+
+			if (state.workflows[workflowId]) {
+				state.workflows[workflowId].active = true;
```
Comment (mutdmour) [inline packages/editor-ui/src/store.ts:209]: Vue.set for reactivity?

File: packages/editor-ui/src/store.ts [hunk 11]
```diff
@@ -185,12 +204,20 @@ export const store = new Vuex.Store({
...
+			}
 		},
 		setWorkflowInactive(state, workflowId: string) {
 			const index = state.activeWorkflows.indexOf(workflowId);
 			if (index !== -1) {
 				state.activeWorkflows.splice(index, 1);
 			}
+
+			if (state.workflows[workflowId]) {
+				state.workflows[workflowId].active = false;
```
Comment (mutdmour) [inline packages/editor-ui/src/store.ts:219]: Vue.set for reactivity?

File: packages/editor-ui/src/components/MainSidebar.vue [hunk 12]
```diff
@@ -326,12 +303,8 @@ export default mixins(
 			},
 			async handleSelect (key: string) {
 				switch (key) {
-					case '/workflow': {
-						await this.createNewWorkflow();
-						break;
-					}
-					case 'workflow-open': {
-						this.$store.dispatch('ui/openModal', WORKFLOW_OPEN_MODAL_KEY);
```
[the remark this answers is not in the document]
Comment (alexgrozav) [reply packages/editor-ui/src/components/MainSidebar.vue:334]: Done.

File: packages/editor-ui/src/components/WorkflowCard.vue (hunk 1 above)
Comment (alexgrozav) [reply packages/editor-ui/src/components/WorkflowCard.vue:199]: Wasn't entirely sure how it works. Thanks for pointing that out! Fixed now.

File: packages/editor-ui/src/components/forms/ResourceFiltersDropdown.vue (hunk 3 above)
Comment (alexgrozav) [reply packages/editor-ui/src/components/forms/ResourceFiltersDropdown.vue:78]: Good point. Done!

File: packages/editor-ui/src/components/forms/ResourceSharingFilterSelect.ee.vue (hunk 4 above)
Comment (alexgrozav) [reply packages/editor-ui/src/components/forms/ResourceSharingFilterSelect.ee.vue]: Migrated to n8n-user-select (added support for disabled entries as well).

File: packages/editor-ui/src/plugins/i18n/locales/en.json (hunk 5 above)
Comment (alexgrozav) [reply packages/editor-ui/src/plugins/i18n/locales/en.json:1139]: Most of them will be needed for the workflow sharing feature. I'll add a reminder to remove what's not needed afterwards if that's ok with you.

File: packages/editor-ui/src/components/layouts/ResourcesListLayout.vue (hunk 6 above)
Comment (alexgrozav) [reply packages/editor-ui/src/components/layouts/ResourcesListLayout.vue]: Good catch! Fixed.

File: packages/editor-ui/src/components/layouts/ResourcesListLayout.vue (hunk 7 above)
Comment (alexgrozav) [reply packages/editor-ui/src/components/layouts/ResourcesListLayout.vue]: Good catch, removed.

File: packages/editor-ui/src/components/layouts/ResourcesListLayout.vue (hunk 9 above)
Comment (alexgrozav) [reply packages/editor-ui/src/components/layouts/ResourcesListLayout.vue]: Done!

File: packages/editor-ui/src/store.ts (hunk 10 above)
Comment (alexgrozav) [reply packages/editor-ui/src/store.ts]: Done!

File: packages/editor-ui/src/store.ts (hunk 11 above)
Comment (alexgrozav) [reply packages/editor-ui/src/store.ts]: Done!

File: packages/editor-ui/src/components/layouts/ResourcesListLayout.vue (hunk 8 above)
Comment (alexgrozav) [reply packages/editor-ui/src/components/layouts/ResourcesListLayout.vue]: Done!

File: packages/design-system/src/components/N8nTags/Tags.vue [hunk 13]
```diff
@@ -1,44 +1,87 @@
...
-		<n8n-tag v-for="tag in tags" :key="tag.id" :text="tag.name" @click="$emit('click', tag.id, $event)"/>
+		<n8n-tag v-for="tag in visibleTags" :key="tag.id" :text="tag.name" @click="$emit('click', tag.id, $event)"/>
+		<n8n-link
+			v-if="truncate && !showAll && hiddenTagsLength > 0"
+			theme="text"
+			underline
+			size="small"
+			@click.stop.prevent="showAll = true"
+		>
+			+{{ hiddenTagsLength }} more
```
Comment (mutdmour) [inline packages/design-system/src/components/N8nTags/Tags.vue:11]: i18n?

File: packages/editor-ui/src/views/NodeView.vue [hunk 14]
```diff
@@ -725,15 +724,6 @@ export default mixins(
 					e.preventDefault();
 
 					this.callDebounced('cutSelectedNodes', { debounceTime: 1000 });
-				} else if (e.key === 'o' && this.isCtrlKeyPressed(e) === true) {
```
Comment (mutdmour) [inline packages/editor-ui/src/views/NodeView.vue:728]: reminder to remind Deb to update docs

File: packages/editor-ui/src/components/layouts/ResourcesListLayout.vue [hunk 15]
```diff
@@ -301,12 +299,12 @@ export default mixins(
...
 		},
 		sendSubviewTelemetry() {
-			this.$telemetry.track('User changed credentials sub view', {
+			this.$telemetry.track(`User changed ${this.resourceKey} sub view`, {
 				sub_view: this.getTelemetrySubview(),
 			});
 		},
 		sendSortingTelemetry() {
-			this.$telemetry.track('User changed sorting in cred list', {
+			this.$telemetry.track(`User changed sorting in ${this.resourceKey} list`, {
```
Comment (mutdmour) [inline packages/editor-ui/src/components/layouts/ResourcesListLayout.vue:307]: reminder to update telemetry table with new event and let david know....

File: packages/editor-ui/src/components/forms/ResourceFiltersDropdown.vue [hunk 16]
```diff
@@ -151,3 +171,8 @@ export default Vue.extend({
 	align-items: center;
 }
 </style>
+
+<style lang="scss" scoped>
```
Comment (mutdmour) [inline packages/editor-ui/src/components/forms/ResourceFiltersDropdown.vue:175]: no closing tag here?

Comment (alexgrozav) [reply packages/editor-ui/src/components/forms/ResourceFiltersDropdown.vue]: Forgot about this. Removed.

File: packages/editor-ui/src/components/layouts/ResourcesListLayout.vue (hunk 15 above)
Comment (alexgrozav) [reply packages/editor-ui/src/components/layouts/ResourcesListLayout.vue:307]: Done. David is aware and agreed to the change.

File: packages/editor-ui/src/views/NodeView.vue (hunk 14 above)
Comment (alexgrozav) [reply packages/editor-ui/src/views/NodeView.vue:728]: Done

File: packages/design-system/src/components/N8nTags/Tags.vue (hunk 13 above)
Comment (alexgrozav) [reply packages/design-system/src/components/N8nTags/Tags.vue]: Good point. Also added story for truncated tags.

File: packages/cli/config/schema.ts [hunk 17]
```diff
@@ -880,6 +880,7 @@ export const schema = {
 			sharing: {
 				format: Boolean,
 				default: false,
+				env: 'N8N_ENTERPRISE_FEATURES_SHARING',
```
Comment (mutdmour) [inline packages/cli/config/schema.ts:883]: I assume you did not mean to push this?

Comment (alexgrozav) [reply packages/cli/config/schema.ts]: That's absolutely right, fixed.

**Review by mutdmour — DISMISSED** [review]

**Review by mutdmour — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.199.0`

**Pull request merged**